platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/EventStreamBound.java

```diff
@@ -0,0 +1,287 @@
+     *
+     * @param event     the event implementation to compare to
+     * @param boundType the type of bound to compare as
+     * @return a negative integer, zero, or a positive integer based on whether the consensus data is less, equal or
+     * greater than the bound.
+     */
+    public Integer compareTo(EventImpl event, BoundType boundType) {
+        return compareTo(event.getConsensusData(), boundType);
+    }
+
+    /**
+     * Compares the round received and consensus timestamp of the consensusData to the bound.  The comparison is made
+     * with the frame of reference of the consensus data. If the data is higher, a value greater than 0 is returned, if
+     * the data is lower, a valued less than 0 is returned, otherwise 0 is returned.
+     * <p>
+     * If the bound is a LOWER bound, if both the round and the timestamp are present in the bound then the consensus
+     * data is greater than bound if both the bound's round and the timestamp are lower than the consensus data round
+     * received and consensus timestamp, the consensus data is less than the bound if either the bound's round or the
+     * timestamp are strictly greater, and equal otherwise.   If only one of the timestamp or round is present, the
+     * normal comparison rules apply. If neither the timestamp nor round is present, the bound is an unbounded lower
+     * bound and the event is always greater than it.
+     * <p>
+     * If the bound is an UPPER bound, if both the round and the timestamp are present then the consensus data is less
+     * than the bound if both the bound's round and the timestamp are greater than the consensus data round received and
+     * consensus timestamp, the consensus data is greater than the bound if either bound's the round or the timestamp
+     * are strictly less, and equal otherwise.   If only one of the timestamp or round is present, the normal comparison
+     * rules apply. If neither the timestamp nor round is present, the bound is an unbounded upper bound and the event
+     * is always less than it.
+     *
+     * @param consensusData the consensus data to compare to
+     * @param boundType     the type of bound
+     * @return A positive value if the consensus data is greater than the bound, A negative value if the consensus data
+     * is less than the bound, and 0 otherwise.
+     */
+    public int compareTo(@NonNull final ConsensusData consensusData, BoundType boundType) {
+        if (isUnbounded()) {
+            return boundType == BoundType.LOWER ? 1 : -1;
+        }
+        if (round == NO_ROUND) {
+            return compareTimestamp(consensusData);
+        }
+        if (timestamp == NO_TIMESTAMP) {
+            return compareRound(consensusData);
+        }
+
+        int roundCompare = compareRound(consensusData);
+        int timestampCompare = compareTimestamp(consensusData);
+
+        if (boundType == BoundType.LOWER) {
```

Comment:
Is it possible to reach this code using the CLI since it prevents both the round and the timestamp from being set?

Reply:
The unit tests test the code, but in production the exception in the EventStreamCommandInfo prevents making this code reachable.

Reply:
I have removed the simultaneous use of round and timestamp.